src/app/_components/BlockList/LayoutA/useBlockListWebSocket.ts

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useRef, useState } from 'react';
+
+import { NakamotoBlock } from '@stacks/blockchain-api-client/src/generated/models';
+import { Block } from '@stacks/stacks-blockchain-api-types';
+
+import { UIBlockType, UISingleBlock } from '../types';
+import { useSubscribeBlocks } from '../useSubscribeBlocks';
+
+export function useBlockListWebSocket(
+  initialBlockHashes: Set<string>,
+  initialBurnBlockHashes: Set<string>
+) {
+  const [latestBlocks, setLatestBlocks] = useState<UISingleBlock[]>([]);
```

Comment:
Why do we need to use state and refs?

Reply:
Refs are used to keep track of values that don't trigger UI updates when they change, unlike state changes, which do cause UI updates.